cmake-modules/AddAzureSDKforC.cmake

```diff
@@ -0,0 +1,25 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+#
+# How to use: Copy this file to a cmake modules folder within your project.
+# - Add `list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/your-cmake-modules-folder")`
+# - Call this module by doing `include(AddAzureSDKforC)` from the root CMakeLists.txt.
+# - Link your application against your CMake targets, for example, `target_link_libraries(yourApp PRIVATE az_core)`.
+#
+# This will create a _deps folder for the generated project and Azure SDK will be checked out in there.
+# When calling `cmake --build .`, Azure SDK will be built before your application so it can be available for your app to consume.
+#
+# CMake options for Azure SDK for C can be set together with your application CMake options.
+#
+# Note: Update GIT_TAG to the expected version from Azure SDK for C
+#
+
+include(FetchContent)
+FetchContent_Declare(azuresdkforc
+    GIT_REPOSITORY      https://github.com/Azure/azure-sdk-for-c.git
+    GIT_TAG             1.0.0-preview.6)
```

Comment:
Future improvement: Try to leverage the **single** version.h file update to update this as well. That update happens automatically as part of our releases. We don't want this to get out-of-sync and we don't want version info floating around in the repo in multiple places.

cc @danieljurek

Reply:
https://github.com/Azure/azure-sdk-for-c/issues/1311